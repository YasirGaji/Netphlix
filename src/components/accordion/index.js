import React, { useState, useContext, createContext } from 'react'
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
  Frame,
} from './styles/accordion';

const ToggleContext = createContext();

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>
}

// Accordion.Item = function AccordionItem({ children, ...restProps }) {
//   const [toggleShow, setToggleShow] = useState(false)

//   return (
//     <Item {...restProps}>
//       <Header onClick={() => setToggleShow((toggleShow) => !toggleShow)}>
//         {children}
//       </Header>
//       <Body className={toggleShow ? 'open' : 'close'}>{children}</Body>
//     </Item>
//   )
// }

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  )
}


Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header onClick={() => setToggleShow((toggleShow) => !toggleShow)} {...restProps}>
      {children}
    </Header>
  )
}


Accordion.Body = function AccordionBody({ children, ...restProps }) {
  return <Body {...restProps}>{children}</Body>
}
