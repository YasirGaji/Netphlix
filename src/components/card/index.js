import React from 'react'
import {
  Container,
  Group,
  Title,
  SubTitle,
  Text,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
  Content,
  Meta,
  Entities,
  Item,
  Image,
} from './styles/card'

export default function Card({ children, ...restProps}) {
  return <Container { ...restProps}>{children}</Container>
}

Card.Group = function CardGroup({ children, ...restProps}) {
  return <Group { ...restProps}>{children}</Group>
}

Card.Title = function CardTitle({ children, ...restProps}) {
  return <Title { ...restProps}>{children}</Title>
}

Card.SubTitle = function CardSubTitle({ children, ...restProps}) {
  return <SubTitle { ...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps}) {
  return <Text { ...restProps}>{children}</Text>
}

Card.Feature = function CardFeature({ children, category, ...restProps}) {
  return <Feature { ...restProps} src={`/images/${category}/${category}.png`}>{children}</Feature>
}

Card.FeatureTitle = function CardFeatureTitle({ children, ...restProps}) {
  return <FeatureTitle { ...restProps}>{children}</FeatureTitle>
}

Card.FeatureText = function CardFeatureText({ children, ...restProps}) {
  return <FeatureText { ...restProps}>{children}</FeatureText>
}

Card.FeatureClose = function CardFeatureClose({ children, ...restProps}) {
  return <FeatureClose { ...restProps}>{children}</FeatureClose>
}

Card.Maturity = function CardMaturity({ children, ...restProps}) {
  return <Maturity { ...restProps}>{children}</Maturity>
}

Card.Content = function CardContent({ children, ...restProps}) {
  return <Content { ...restProps}>{children}</Content>
}

Card.Meta = function CardMeta({ children, ...restProps}) {
  return <Meta { ...restProps}>{children}</Meta>
}

Card.Entities = function CardEntities({ children, ...restProps}) {
  return <Entities { ...restProps}>{children}</Entities>
}

Card.Item = function CardItem({ item, children, ...restProps}) {
  const { genre } = item
  return (
    <Item { ...restProps}>
      <Image src={`/images/${genre}/${item.slug}/small.jpg`} />
      {children}
    </Item>
  )
}

Card.Image = function CardImage({ ...restProps}) {
  return <Image { ...restProps} />
}

