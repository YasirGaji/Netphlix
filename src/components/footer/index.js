import React from 'react'
import { 
  Container,
  Row,
  Column,
  Link,
  Title,
  Text,
  Break
} from './styles/footer'

export default function Footer({ children, ...restProps}) {
  return (
    <Container {...restProps}>{children}</Container>
  )
}


Footer.Row = function FooterRow({ children, ...restProps}) {
  return <Row {...restProps}>{children}</Row>
}

Footer.Column = function FooterRow({ children, ...restProps}) {
  return <Column {...restProps}>{children}</Column>
}

Footer.Link = function FooterRow({ children, ...restProps}) {
  return <Link {...restProps}>{children}</Link>
}

Footer.Title = function FooterRow({ children, ...restProps}) {
  return <Title {...restProps}>{children}</Title>
}

Footer.Text = function FooterRow({ children, ...restProps}) {
  return <Text {...restProps}>{children}</Text>
}

Footer.Break = function FooterRow({ children, ...restProps}) {
  return <Break {...restProps}>{children}</Break>
}