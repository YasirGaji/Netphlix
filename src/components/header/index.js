import React from 'react'
import { 
  Background, 
  Container, 
  Logo, 
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Search,
  SearchIcon,
  SearchInput,
  Picture,
  Profile,
  Dropdown,
} from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'


export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
  const [searchActive, setSearchActive] = React.useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(searchActive => !searchActive)}>
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series by yasir"
        active={searchActive}
      />
    </Search>
  );
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>; 
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
}