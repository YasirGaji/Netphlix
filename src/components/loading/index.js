import React from "react";
import { Container, Picture } from "./styles/loading";


export const Loading = ({ src, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Picture src={`/images/users/${src}.png`} />
    </Container>
  );
}

