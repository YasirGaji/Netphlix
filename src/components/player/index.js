import React, { useState, useContext, createContext } from "react";


