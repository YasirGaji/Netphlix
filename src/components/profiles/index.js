import React from "react";
import { 
  Container, 
  Title, 
  List, 
  User, 
  Picture,
  Name 
} from "./styles/profiles";

export default function Profiles({children, ...restProps}) {
  return <Container {...restProps}>{children}</Container>
}

Profiles.Title = function ProfilesTitle({children, ...restProps}) {
  return <Title {...restProps}>{children}</Title>
}

Profiles.List = function ProfilesList({children, ...restProps}) {
  return <List {...restProps}>{children}</List>
}

Profiles.User = function ProfilesUser({children, ...restProps}) {
  return <User {...restProps}>{children}</User>
}

Profiles.Picture = function ProfilesPicture({src, ...restProps}) {
  return <Picture {...restProps} src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`} />
}

Profiles.Name = function ProfilesName({children, ...restProps}) {
  return <Name {...restProps}>{children}</Name>
}

