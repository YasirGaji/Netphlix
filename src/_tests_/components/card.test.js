/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render } from "@testing-library/react";
import { Card, Player } from "../../components";
// import { cardData } from "../../fixtures/cardData.json";

const category = "series";
const slideRows = [
  {
    title: "Documentaries",
    data: [
      {
        description: "Tiger King description",
        docId: "1",
        genre: "documentaries",
        id: "1",
        maturity: "18",
        slug: "tiger-king",
        title: "Tiger King",
      },
    ],
  },
  {
    title: "Comedies",
    data: [
      {
        description: "The Office description",
        docId: "2",
        genre: "comedies",
        id: "2",
        maturity: "15",
        slug: "the-office",
        title: "The Office",
      },
    ],
  },
];

describe("<Card />", () => {
  it("renders the <Card /> with populated data", () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/thanks.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(getByText("Documentaries")).toBeTruthy();
    expect(getByText("Tiger King")).toBeTruthy();
    expect(getByText("Tiger King description")).toBeTruthy();

    expect(getByText("Comedies")).toBeTruthy();
    expect(getByText("The Office")).toBeTruthy();
    expect(getByText("The Office description")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
 
  });
});