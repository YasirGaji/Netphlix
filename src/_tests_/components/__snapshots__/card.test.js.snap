// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the <Card /> with populated data 1`] = `
<div
  class="card__Group-sc-1vhzc8k-2 innKtm"
>
  <div
    class="card__Container-sc-1vhzc8k-1 BdCgl"
  >
    <p
      class="card__Title-sc-1vhzc8k-0 bUzBVU"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-1vhzc8k-5 dDznZT"
    >
      <div
        class="card__Item-sc-1vhzc8k-8 fpYDAo"
      >
        <img
          class="card__Image-sc-1vhzc8k-7 kiMkc"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-1vhzc8k-6 jUwoxn"
        >
          <p
            class="card__SubTitle-sc-1vhzc8k-3 NUeTI"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-1vhzc8k-4 eIQlrF"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-1vhzc8k-1 BdCgl"
  >
    <p
      class="card__Title-sc-1vhzc8k-0 bUzBVU"
    >
      Comedies
    </p>
    <div
      class="card__Entities-sc-1vhzc8k-5 dDznZT"
    >
      <div
        class="card__Item-sc-1vhzc8k-8 fpYDAo"
      >
        <img
          class="card__Image-sc-1vhzc8k-7 kiMkc"
          src="/images/series/comedies/the-office/small.jpg"
        />
        <div
          class="card__Meta-sc-1vhzc8k-6 jUwoxn"
        >
          <p
            class="card__SubTitle-sc-1vhzc8k-3 NUeTI"
          >
            The Office
          </p>
          <p
            class="card__Text-sc-1vhzc8k-4 eIQlrF"
          >
            The Office description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
